datadog/resource_datadog_dashboard.go

```diff
@@ -315,6 +328,10 @@ func buildDatadogDashboard(d *schema.ResourceData) (*datadogV1.Dashboard, error)
 	if v, ok := d.GetOk("is_read_only"); ok {
 		dashboard.SetIsReadOnly(v.(bool))
 	}
+	if v, ok := d.GetOk("restricted_roles"); ok && !dashboard.GetIsReadOnly() {
```

Comment:
We probably don't need `!dashboard.GetIsReadOnly()` once the fields are set to conflict with one another in the schema.

Reply:
This one is necessary, for 1 specific case: when we're trying to set `is_read_only: true`. The issue is that if the API receives both `is_read_only` + `restricted_roles` on a request, the `restricted_roles` will take priority -- this is an issue in the case where a user has `is_read_only = true` in their config.